OIPA/iati/parser/tests/test_2_03_parser.py

```diff
@@ -2095,3 +2095,136 @@ def test_activity_result_indicator_document_link_document_category(self):
                          document_link)
         self.assertEqual(indicator_document_link_category.category,
                          indicator_document_category)
+
+
+class ActivityResultReferenceTestCase(TestCase):
+    '''
+    2.03: Added new (optional) <reference> element for <result>
+    element.
+    '''
+
+    def setUp(self):
+        # 'Main' XML file for instantiating parser:
+        xml_file_attrs = {
+            "generated-datetime": datetime.datetime.now().isoformat(),
+            "version": '2.03',
+        }
+        self.iati_203_XML_file = E("iati-activities", **xml_file_attrs)
+
+        dummy_source = synchroniser_factory.DatasetFactory.create(
+              name="dataset-2"
+        )
+
+        self.parser_203 = ParseManager(
+             dataset=dummy_source,
+             root=self.iati_203_XML_file,
+        ).get_parser()
+
+        # Related objects:
+        # create dummy object
+        self.result_vocabulary = ResultVocabularyFactory(code='99')
+        self.result = iati_factory.ResultFactory.create()
+
+        self.parser_203.register_model('Result', self.result)
+
+    def test_activity_result_reference(self):
+        """
+        Test if result, code, vocabulary_uri attributes in <reference> XML
+        element is correctly saved.
+        """
+
+        # case 1: where 'vocabulary' is missing
+
+        result_reference_attr = {
+            # "vocabulary": result_vocabulary.code,
+            "code": '01',
+            "vocabulary-uri": 'www.example.com'
+
+        }
+        result_reference_XML_element = E(
+            'result-reference',
```

Comment:
According to the [IATI docs for this element](http://reference.iatistandard.org/203/activity-standard/iati-activities/iati-activity/result/reference/), it is called `<reference>` and not `<result-reference>` :)